compiler/src/dotty/tools/dotc/transform/PCPCheckAndHeal.scala

```diff
@@ -175,8 +175,10 @@ class PCPCheckAndHeal(@constructorOnly ictx: Context) extends TreeMapWithStages(
       tp match
         case tp: TypeRef =>
           tp.prefix match
-            case NoPrefix if level > levelOf(tp.symbol) =>
-              tryHeal(tp.symbol, tp, pos)
+            case NoPrefix if level > levelOf(tp.symbol) && !tp.typeSymbol.hasAnnotation(defn.InternalQuoted_QuoteTypeTagAnnot) =>
+              val tp1 = tp.dealias
```

Comment:
What about making `tp.dealias` as the scrutinee of the top-level pattern match?

Reply:
Done